docs/data/date-pickers/custom-opening-button/AddWarningIconWhenInvalidRange.tsx

```diff
@@ -12,7 +12,18 @@ function CustomInputAdornment(props: InputAdornmentProps & { hasError?: boolean
   const { hasError, children, sx, ...other } = props;
   return (
     <InputAdornment {...other}>
-      <PriorityHighIcon color="error" sx={{ opacity: hasError ? 1 : 0 }} />
+      <PriorityHighIcon
+        color="error"
+        sx={[
+          hasError
+            ? {
+                opacity: 1,
+              }
+            : {
+                opacity: 0,
+              },
+        ]}
+      />
```

Comment:
Is this change required? It does not seem to make any difference, except that it now creates 2 objects instead of one 🤔

Reply:
If I remember correctly, @siriwatknp confirmed that is is required for Pigment CSS to work, the previous syntax would not be supported by static extraction. 🤔

Reply:
This is required. Please have a look at my [explanation](https://github.com/mui/mui-x/pull/14143/files/e21fdcf9baa322e13ba67c5c8c868c5b9287bf2c#r1723650800)

Reply:
By the way, in cases such as this one in which there's one condition, you can remove the array

```js
        sx={
          hasError
            ? {
                opacity: 1,
              }
            : {
                opacity: 0,
              },
        }
```